src/metabase/domain_entities/card.cljc

```diff
@@ -0,0 +1,260 @@
+(def VisualizationSettings
+  "Malli schema for `visualization_settings` - a map of strings to opaque JS values."
+  [:map-of string? :any])
+
+(def LocalFieldReference
+  "Malli schema for a *legacy* field clause for a local field."
+  [:tuple
+   [:= :field]
+   number?
+   [:maybe [:map-of string? :any]]])
+
+(def ForeignFieldReference
+  "Malli schema for a *legacy* field clause for a foreign field."
+  [:tuple
+   [:= :field]
+   [:or number? string?]
+   [:map [:source-field {:js/prop "source-field"} [:or number? string?]]]])
+
+(def VariableTarget
+  "Malli schema for a parameter that references a template variable."
+  [:tuple [:= :template-tag] string?])
+
+(def ParameterTarget
+  "Malli schema for a parameter's target field."
+  [:orn
+   [:variable [:tuple [:= :variable] VariableTarget]]
+   [:dimension [:tuple
+                [:= :dimension]
+                [:or LocalFieldReference ForeignFieldReference VariableTarget]]]])
+
+(def Parameter
+  "Malli schema for each Card.parameters value."
+  [:map
+   [:id string?]
+   [:name string?]
+   [:display-name {:optional true :js/prop "display-name"} string?]
+   [:slug string?]
+   [:type string?]
+   [:sectionId {:optional true :js-prep "sectionId"} string?]
+   [:default {:optional true} :any]
+   [:filtering-parameters {:optional true :js/prop "filteringParameters"} [:vector string?]]
+   [:is-multi-select {:optional true :js/prop "isMultiSelect"} boolean?]
+   [:required {:optional true} boolean?]
+   [:target {:optional true} ParameterTarget]
+   [:value {:optional true} :any]
+   [:values_query_type {:optional true} :any]
+   [:values_source_config {:optional true} :any]
+   [:values_source_type {:optional true} :any]])
+
+(def Card
+  "Malli schema for a possibly-saved Card."
+  [:schema
+   [:map
+    [:archived {:optional true} boolean?]
+    [:average_query_time {:optional true} number?]
+    [:breakout-column {:optional true :js/prop "_breakoutColumn"} :any]
+    [:breakout-value {:optional true :js/prop "_breakoutValue"} :any]
+    [:cache-ttl {:optional true} [:maybe number?]]
+    [:can_write {:optional true} boolean?]
+    [:collection {:optional true} :any]
+    [:collection_id [:maybe number?]]
+    [:collection_position [:maybe number?]]
+    [:collection_preview boolean?]
+    [:created_at [:maybe string?]] ;; TODO: Date regex?
+    [:creator {:optional true}
+     [:map
+      [:id number?]
+      [:common_name string?]
+      [:date_joined string?] ;; TODO: Date regex
+      [:email string?]
+      [:first_name  [:maybe string?]]
+      [:is_qbnewb boolean?]
+      [:is_superuser boolean?]
+      [:last_login string?] ;; TODO: Date regex
+      [:last_name   [:maybe string?]]]]
+    [:creator_id {:optional true} number?]
+    [:creation-type {:optional true :js/prop "creationType"} string?]
+    [:dashboard_count {:optional true} number?]
+    [:dashboard-id {:optional true :js/prop "dashboardId"} number?]
```

Comment:
It seems a little bit weird to mix and match kebab-case and snake_case, if we're already converting JS names from JS to kebeb-case, can we also convert names in Clojure-land? If not I'm worried that this would be hard to use in JVM Clojure, because we're still using `:dashboard_id` there

Reply:
The mixing of names here is really unfortunate, I agree. There's issues with schemes to try to clean this up, though.

- This isn't actually the `dashboard_id` field from the appdb. It's attached dynamically in the client sometimes to keep track of the parent dashboard. (Hence the JS-convention name `dashboardId`.)
    - In practice it's not even used, it's just there so the conversion of `Card`s is done correctly in JS-land.
- There's not currently any point where conversion is happening in Clojure-land.

That said, for fields like this that we don't actually expect to see on the BE side right now, we could spell this specific one as `:database_id` just for minimal surprise.